openedx/features/qverse_features/registration/signals.py

```diff
@@ -48,30 +47,31 @@ def create_users_from_csv_file(sender, instance, created, **kwargs):
     # whitespaces. So, we will have to handle that case ourselves
     reader = (dict((k.strip().lower(), v.strip() if v else v) for k, v in row.items()) for row in dict_reader)
     output_file_rows = []
-    output_row = {}
     try:
         for row in reader:
-            try:
-                row.update({'status': ''})
-                _validate_single_csv_row(row)
-                # Creating/Updating edX User, edX User Profile and QVerse User Profile
-                # will be an atomic operation. If one operation fails the previous
-                # successfull operations will be reverted and no changes will be applied.
-                with transaction.atomic():
-                    edx_user = _create_or_update_edx_user(row)
-                    _create_or_update_edx_user_profile(edx_user)
-                    _create_or_update_qverse_user_profile(edx_user, row)
-            except (IntegrityError, ValueError):
-                LOGGER.exception('Error while creating/updating ({}) user and its profiles.'.format(row.get('regno')))
+            row['status'] = ''
+            if _validate_single_csv_row(row):
+                try:
+                    # Creating/Updating edX User, edX User Profile and QVerse User Profile
+                    # will be an atomic operation. If one operation fails the previous
+                    # successfull operations will be reverted and no changes will be applied.
+                    with transaction.atomic():
+                        edx_user = _create_or_update_edx_user(row)
+                        _create_or_update_edx_user_profile(edx_user)
+                        _create_or_update_qverse_user_profile(edx_user, row)
+                except IntegrityError as error:
+                    LOGGER.exception('Error while creating/updating ({}) user and its '
+                                     'profiles with the following errors {}.'.format(row.get('regno'), error))
+                    row.update({'status': 'Failed'})
```

Comment:
no need to update you can simply reassigned value `row['status'] = 'Failed'`

Reply:
done.